### PR TITLE
Use date modified as section identifier for drafts

### LINKED ISSUE #6476: Draft pages missing
A user reports that all draft pages are missing from the app. After looking at the logs and some investigation we have this:

```
CoreData: error: (NSFetchedResultsController) The fetched object at index 2 has an out of order section name 'last week. Objects must be sorted by section name'
2017-01-26 17:07:42:492 WordPress[2148:4614655] Error fetching posts after updating the fetch request predicate: Error Domain=NSCocoaErrorDomain Code=134060 "(null)" UserInfo={reason=The fetched object at index 2 has an out of order section name 'last week. Objects must be sorted by section name'}
```

In other words, the `sortDescriptiors` and `sectionNameKeyPath` of a `NSFetchedResultsController` need to be based on the same field. This is the case, most of the times, but for drafts we are using `dateModified` instead of `date_created_gmt`, which is what we are using to build the section labels.

### Steps to reproduce the behavior

What you need to do is get into a state where there is a conflict between the section name and the sorting:

- Page A (Today)
- Page B (Last week)
- Page C (Today)

Section name is based on the date you specify for the page, and sort order is based on the actual date when you made the last change, so try this in a browser:

1. Create a draft page (Page A), set the date to 1 hour ago
1. Create a draft page (Page B), set the date to 10 days ago
1. Create a draft page (Page C), set the date to 2 hours ago
1. Open the app and go to the pages list, switch to drafts
1. Pull to refresh, the list should be empty

Note that if you edit Page B and save it, pages will appear again, but the sections will have the wrong order.

simulator screen shot 26 jan 2017 17 19 18

### PR DESCRIPTION
**Pull request by koke**
Fixes #6476 

As I was writing the commit message, I realized there were two options here:

- Using the modified date for the section labels on draft pages (what I did)
- Sort drafts by modified date (#5558) only for posts, and sort draft pages by date created

I feel like the first option might be more useful for drafts, since the last edited pages will bubble up to the top of the list, but it's up for discussion

To test:

Follow the steps in #6476 and verify that the draft pages show up in the list

Needs review: @aerych

### CODE REVIEW

**Comment by aerych** [comment]
> I feel like the first option might be more useful for drafts, since the last edited pages will bubble up to the top of the list, but it's up for discussion

Calypso sorts the Page list by date modified so I think that's fine.   However, it would be nice if we could match the sectioning.  Notice that after following the steps described in the issue Calypso shows three sections with one row per section, but the app lumps all three rows in a singe section.

screen shot 2017-01-30 at 2 25 00 pm


simulator screen shot jan 30 2017 2 26 14 pm

**Comment by aerych** [comment]
Re: the examples above.  Notice that there are two "Today" sections in Calypso.  This is what happens on the web when sorting by modified date but sectioning by date created.  Looks a bit weird and I'm a little skeptical that this would work well with the way an NSFetchedResultsController would handle sectioning.  Do you think it would be in scope for this PR to investigate the possibility or something for a follow up issue?

**Comment by koke** [comment]
> Notice that there are two "Today" sections in Calypso

I do believe calypso is wrong in this case, and agree with `NSFetchedResultsController` in that sectioning and sorting should be based on the same property.

File: WordPress/Classes/Extensions/NSDate+Helpers.swift [hunk 1]
```diff
@@ -20,6 +20,16 @@ extension Date {
             formatter.timeZone      = TimeZone(secondsFromGMT: 0)
             return formatter
         }()
+
+        static let pageSectionFormatter: TTTTimeIntervalFormatter = {
```
Comment (koke) [inline WordPress/Classes/Extensions/NSDate+Helpers.swift]: I think the goal was having all the formatters in the same place, so it's easier to spot all the different ways we're showing dates across the app.

**Pull request merged**